From dcc225ce4d200e49beb4fe4e5b96d58ca22f295d Mon Sep 17 00:00:00 2001
From: Mingliang Liu <liuml07@apache.org>
Date: Thu, 23 Mar 2017 11:33:06 -0700
Subject: [PATCH 2356/2848] HADOOP-14214. DomainSocketWatcher::add()/delete()
 should not self interrupt while looping await().
 Contributed by Mingliang Liu

(cherry picked from commit d35e79abc2fee7153a6168e6088f100de59d8c81)
(cherry picked from commit 1e64c582f5a6ceec2ebecaa789a19158a1e5ecfb)

Change-Id: I39e30ad11ffd58cd7a542d29b544f5e2a5a554e9
---
 .../hadoop/net/unix/DomainSocketWatcher.java       |   12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/unix/DomainSocketWatcher.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/unix/DomainSocketWatcher.java
index f64670d..ff25e18 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/unix/DomainSocketWatcher.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/unix/DomainSocketWatcher.java
@@ -319,11 +319,7 @@ public void add(DomainSocket sock, Handler handler) {
       toAdd.add(entry);
       kick();
       while (true) {
-        try {
-          processedCond.await();
-        } catch (InterruptedException e) {
-          Thread.currentThread().interrupt();
-        }
+        processedCond.awaitUninterruptibly();
         if (!toAdd.contains(entry)) {
           break;
         }
@@ -345,11 +341,7 @@ public void remove(DomainSocket sock) {
       toRemove.put(sock.fd, sock);
       kick();
       while (true) {
-        try {
-          processedCond.await();
-        } catch (InterruptedException e) {
-          Thread.currentThread().interrupt();
-        }
+        processedCond.awaitUninterruptibly();
         if (!toRemove.containsKey(sock.fd)) {
           break;
         }
-- 
1.7.9.5

